modules/core/src/minmax.cpp

```diff
@@ -1511,8 +1511,27 @@ void cv::minMaxIdx(InputArray _src, double* minVal,
     Mat src = _src.getMat(), mask = _mask.getMat();
 
     if (src.dims <= 2)
-        CALL_HAL(minMaxIdx, cv_hal_minMaxIdx, src.data, src.step, src.cols, src.rows, src.depth(), minVal, maxVal,
-                 minIdx, maxIdx, mask.data);
+    {
+        if (cn == 1)
```

Comment:
since you have a comprehensive CV_Assert() above, there is no need in extra branch. Pass mask.data in both cases, pass `minIdx ? minIdx : &_minIdx`, `maxIdx ? maxIdx : &_maxIdx` in both cases, pass `src.cols*cn` in both cases.